### PR TITLE
mpvScripts.{blacklistExtensions, seekTo}: unstable-2022-10-02 → 2023-11-25

### PR DESCRIPTION
**Pull request by nicoonoclaste**
## Description of changes

In `mpv/scripts/occivink.nix`:
- ~~added custom `updateScript`~~
  somehow got `common-updater-scripts` to cope with this kind of factored-out drvs
- ran `updateScript`, updating from `unstable-2022-10-02` to `2023-11-25`
  fix a bug in `blacklistExtensions`, breaking the `remove_files_without_extension` option

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [x] Tested `updateScript`
- [x] Tested compilation of all affected packages
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/applications/video/mpv/scripts/occivink.nix [hunk 1]
```diff
@@ -1,27 +1,52 @@
...
+        set -eux
+        PATH="${with lib; concatStringsSep ":" (map (drv: getBin drv + "/bin") [ curl jq nix-prefetch-github ])}:$PATH"
+
+        oldRev="$(jq -r .rev < ${toString ./occivink.json})"
+        owner="$(jq -r .owner < ${toString ./occivink.json})"
+        repo="$(jq -r .repo < ${toString ./occivink.json})"
+
+        prefetch="$(nix-prefetch-github --json $owner $repo)"
+        rev="$(jq -r .rev <<< "$prefetch")"
+        [ "$rev" != "$oldRev" ] || exit 0
```
Comment (pbsds) [inline pkgs/applications/video/mpv/scripts/occivink.nix:41]:
```suggestion
        [ "$rev" != "$oldRev" ] || {
          >&2 echo There is no new version 
          exit 0
        }
```

File: pkgs/applications/video/mpv/scripts/occivink.nix [hunk 2]
```diff
@@ -33,13 +58,7 @@ in
...
-    outputHash = "sha256-3RlbtUivmeoR9TZ6rABiZSd5jd2lFv/8p/4irHMLshs=";
-  };
-
-  blacklistExtensions = {
-    meta.description = "Automatically remove playlist entries based on their extension.";
-    outputHash = "sha256-qw9lz8ofmvvh23F9aWLxiU4YofY+YflRETu+nxMhvVE=";
-  };
+  seekTo.meta.description = "Mpv script for seeking to a specific position";
+  blacklistExtensions.meta.description =
+    "Automatically remove playlist entries based on their extension.";
```
Comment (pbsds) [inline pkgs/applications/video/mpv/scripts/occivink.nix:63]: was `outputHash` never needed?

File: pkgs/applications/video/mpv/scripts/occivink.nix [hunk 3]
```diff
@@ -1,27 +1,55 @@
...
 let
+  inherit (builtins) toString;
+  inherit (lib) getAttrs recursiveUpdate;
+
   camelToKebab = let
     inherit (lib.strings) match stringAsChars toLower;
     isUpper = match "[A-Z]";
   in stringAsChars (c: if isUpper c != null then "-${toLower c}" else c);
 
+  repoMetadata = lib.importJSON ./occivink.json;
```
Comment (mkg20001) [inline pkgs/applications/video/mpv/scripts/occivink.nix:20]: why are you loading it as json?

File: pkgs/applications/video/mpv/scripts/occivink.nix [hunk 4]
```diff
@@ -10,24 +11,30 @@ let
...
 
       meta = with lib; {
         homepage = "https://github.com/occivink/mpv-scripts";
         license = licenses.unlicense;
         maintainers = with maintainers; [ nicoo ];
+
+        # Sad hack needed to make `common-updaters` work here
+        position = with builtins;
+          let pos = unsafeGetAttrPos "version" self;
+          in "${pos.file}:${toString pos.line}";
```
Comment (pbsds) [inline pkgs/applications/video/mpv/scripts/occivink.nix:35]: It is possible to set `pos` passed to `mkDerivation` instead?

https://github.com/NixOS/nixpkgs/blob/bd464981e07c97f47585366a47f7cd35e7f7726c/pkgs/stdenv/generic/make-derivation.nix#L174-L179

**Comment by pbsds** [comment]
Result of `nixpkgs-review pr 272108` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>mpvScripts.blacklistExtensions</li>
    <li>mpvScripts.seekTo</li>
  </ul>
</details>

**Pull request merged**